Tweet-311-Denver/tweet_311_denver_ui
# Api call

### What does this PR do? 

- Adds the apiCall to post a report.
- Creates a loading component.
- Adds conditional rendering to Tweet to display the Loading component when the fetch is pending. 
- Fixes several minor bugs and adds several helper functions to facilitate the fetch. 

---------

### Dependencies: 

- This PR adds `react-native-dotenv` for the secret key. 

---------

### How to manually test: 

- `git fetch`
- `git checkout api-call`
- `npm i` (skipping this will cause the API to fail)
- `touch .env` (installing the env dependency is supposed to generate this automatically but it didn't for me. Make sure it wasn't created before adding.)
- Inside `.env` add `SERVICE_KEY='key goes here'` (skipping this will also make the api call fail)
- `expo start`

Note: The app is currently set up with the **real** post url. If you want to test in a safer environment change the URL to include the test path. 

- Happy path: 
     - Fill out the form with valid info and submit. If the post was successful you should be 
       redirected to Tweet. 
     - Click through Tweet and Success back to home. 
     - Submit another form. Ensure that the loader renders and submits the second form. 

- Sad path: 
     - In `Form.js` remove `email` from line 74 and the comma on the line above. 
     - Fill out a valid form and submit.
     - You should be redirected back to the form once the post fails. Ensure that the text you 
       previously entered loads in the inputs and that the error message displays. 

I think that's it, have fun!
